Marlin/src/inc/Conditionals_LCD.h

```diff
@@ -26,6 +26,11 @@
  * Conditionals that need to be set before Configuration_adv.h or pins.h
  */
 
+
+#if BUTTONS_EXIST(EN1, EN2, ENC)
+  #define HAS_ROTARY_ENCODER 1
+#endif
```

Comment:
This actually can't work because `pins.h` hasn't been included at this point, a common issue to be aware of.